qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -380,38 +387,34 @@ def construct_circuit(
         psi_x = self._feature_map.assign_parameters(x_dict)
         qc.append(psi_x.to_instruction(), qc.qubits)
 
-        if not is_statevector_sim:
-            if y is not None and len(y) != self._feature_map.num_parameters:
-                raise ValueError(
-                    "y and class feature map incompatible dimensions.\n"
-                    f"y has {len(y)} dimensions, but feature map has {self._feature_map.num_parameters}."
-                )
+        return qc
 
-            if y is None:
-                y = x
+    # just a synonym
+    def _construct_circuit_statevector(self, x: ParameterVector, y: ParameterVector = None):
+        """This is just a synonym for ``_construct_circuit_with_feature_map``"""
+        return self._construct_circuit_with_feature_map(x, y)
```

Comment:
Why do we need a synonym?

Reply:
This is a good question, there's a bunch of methods like `whatever()` and `whatever_statevector()` now. But `_construct_circuit_with_feature_map()` is used to construct circuits in both cases: statevector and non-statevector. That's why it is named like now. I can rename it to `_construct_circuit_statevector()`, but then it would be used in `construct_circuit()` that is used in the non-statevector case and such code might be confusing. I hope I did not bring more confusion.

Reply:
I think in this case it would be less confusing to just have one `_construct_circuit_with_feature_map()` method that is used for both statevector and hardware/qasm-simulator. At least when glancing over the code this synonym method confused me at first. At the places where the logic differs it is definitely nice to separate the code into separate methods.

Reply:
@gentinettagian I'm not saying I'm against the change, I wonder what is exactly confusing as method name consistency is also important.